views/default/elgg/spinner.js

```diff
@@ -4,21 +4,34 @@ define(function (require) {
 	var active = false;
 	var SHOW_DELAY = 20;
 
-	$('body').append('<div class="elgg-spinner"><div class="elgg-ajax-loader"></div></div>');
+	$('body').append('<div class="elgg-spinner"><div class="elgg-ajax-loader"></div><div class="elgg-spinner-text elgg-subtext"></div></div>');
 
 	return {
-		start: function () {
+		start: function (text) {
 			active = true;
+			
+			this.clearText();
+			
 			setTimeout(function () {
 				if (active) {
 					$('body').addClass('elgg-spinner-active');
 				}
 			}, SHOW_DELAY);
+			
+			this.setText(text);
 		},
 
 		stop: function () {
 			active = false;
-			$('body').removeClass('elgg-spinner-active');
-		}
+			$('body').removeClass('elgg-spinner-active').find();
+		},
+		
+		setText: function (text) {
+			$('.elgg-spinner .elgg-spinner-text').html(text);
```

Comment:
For XSS mitigation, either use text() to set the text, or rename the argument HTML so users are clear that they need to escape manually.